modal/io_streams.py

```diff
@@ -176,34 +177,21 @@ async def _consume_container_process_stream(self):
         if self._stream_type == StreamType.DEVNULL:
             return
 
-        completed = False
-        retries_remaining = 10
-        while not completed:
-            try:
-                iterator = _container_process_logs_iterator(self._object_id, self._file_descriptor, self._client)
+        def item_handler(item: Optional[bytes]):
+            if self._stream_type == StreamType.STDOUT and item is not None:
+                print(item.decode("utf-8"), end="")
+            elif self._stream_type == StreamType.PIPE:
+                self._container_process_buffer.append(item)
 
-                async for message in iterator:
-                    if self._stream_type == StreamType.STDOUT and message:
-                        print(message.decode("utf-8"), end="")
-                    elif self._stream_type == StreamType.PIPE:
-                        self._container_process_buffer.append(message)
-                    if message is None:
-                        completed = True
-                        break
+        def completion_check(item: Optional[bytes]):
+            return item is None
 
-            except (GRPCError, StreamTerminatedError, ClientClosed) as exc:
-                if retries_remaining > 0:
-                    retries_remaining -= 1
-                    if isinstance(exc, GRPCError):
-                        if exc.status in RETRYABLE_GRPC_STATUS_CODES:
-                            await asyncio.sleep(1.0)
-                            continue
-                    elif isinstance(exc, StreamTerminatedError):
-                        continue
-                    elif isinstance(exc, ClientClosed):
-                        # If the client was closed, the user has triggered a cleanup.
-                        break
-                raise exc
+        iterator = _container_process_logs_iterator(self._object_id, self._file_descriptor, self._client)
```

Comment:
Mentioned this on Slack, but this will only make one RPC ever, which is not what we want. We want to make a new call in every iteration of the loop.